src/divide_conquer_functions/conquer.jl

```diff
@@ -92,8 +94,42 @@ function get_predicates(grammar::AbstractGrammar,
 )::Vector{RuleNode}
 	# We get the first grammar rule that has the specified `sym_constraint` and add constraint to grammar. 
 	clearconstraints!(grammar)
-	idx_rule = grammar.bytype[sym_constraint][1]
-	addconstraint!(grammar, Contains(idx_rule))
+	# Create DomainRuleNode that contains all rules of type sym_constraint and add constraint to grammar
+	rules = grammar.bytype[sym_constraint]
+	domain = DomainRuleNode(grammar, rules)
+	addconstraint!(grammar, ContainsSubtree(domain))
+
+	iterator = BFSIterator(grammar, sym_bool)
+	predicates = Vector{RuleNode}()
+
+	for (i, candidate_program) ∈ enumerate(iterator)
+		candidate_program = freeze_state(candidate_program)
+		push!(predicates, candidate_program)
+		if i >= n_predicates
+			break
+		end
+	end
+	return predicates
```

Comment:
You can replace these lines with a call to the other signature of `get_predicates` below.

Reply:
I think this is resolved with above.